### ISSUE TITLE
List repositories for a new external service / code host

### ISSUE THREAD

**Issue by gl-srgr** [sourcegraph/sourcegraph#46648]
I'll refer to code host as External Service in this issue.

Currently one or more repositories are added by an admin either at the time of external service creation or at the time of external service update. 

For setup experience one goal is to show repositories associated with and made available by an external service, at the time of external service setup.

Acceptance criteria: Frontend can have available the list of repositories to choose from, with minimal input from user (should be just token and external service type).

**Comment by vovakulikov** [comment]
@gl-srgr I think it also would make a lot of sense if we provide ability to select all repositories from user code host organisation. 

For example, I'm a member of companyN and this companyN has GitHub organisation, I want to try Sourcegraph on all repositories from this organization (if there are many repositories it may take a while to pick all repositories manually through picker) 

We probably already have mutation for this, config in code host adding mutation has organization list but we Aldo need to query this organisations on the frontend. So it might be separate issue though.

**Comment by gl-srgr** [comment]
@vovakulikov Showing all repos for a given organization would be a good feature for the march mvp. I believe the main limitation will be with the token that the user provides, which could be limited to certain repos and not be organization wide. I'm reviewing our current logic to see what we can reuse for getting the most exhaustive list of repositories for the given access token

**Comment by vovakulikov** [comment]
@gl-srgr yeah if it's not possible because of token access rules there is not much we can do. 

>Showing all repos for a given organization would be a good feature for the march mvp

Just to be clear I meant that we probably could list just organisations (not all repos within organisations) You can check the current code host management there is a button for including all repositories from particular set of organisations 

<img width="430" alt="Screenshot 2023-01-18 at 22 13 21" src="https://user-images.githubusercontent.com/18492575/213332484-a4c7a7c5-9665-4bad-ad31-8ebe43c1f28a.png">

I was thinking maybe we could have similar logic, a UI where you can to select some organisations and repositories together 

![Untitled-2023-01-05-2131-2](https://user-images.githubusercontent.com/18492575/213333089-6925e1ed-c27c-465e-9ec3-debf23510cba.png)

**Comment by gl-srgr** [comment]
Updated to be specific to github. As we discussed earlier there are discrepancies in the way we query for repos and in the way repos are organized, depending on which Kind of code host it is (e.g. between github, gitlab, bitbucket we have differencs like repos under teams vs. repos under orgs vs. repos mapped to user vs. repos under projects).
